localization/ndt_scan_matcher/include/ndt_scan_matcher/tree_structured_parzen_estimator.hpp

```diff
@@ -0,0 +1,86 @@
+// Copyright 2023 Autoware Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NDT_SCAN_MATCHER__TREE_STRUCTURED_PARZEN_ESTIMATOR_HPP_
+#define NDT_SCAN_MATCHER__TREE_STRUCTURED_PARZEN_ESTIMATOR_HPP_
+
+/*
+A implementation of tree-structured parzen estimator
+Search num_variables double variables in (-1, 1)
+*/
+
+#include <cstdint>
+#include <random>
+#include <vector>
+
+class TreeStructuredParzenEstimator
+{
+public:
+  struct Input
+  {
+    double x, y, z, roll, pitch, yaw;
+    Input operator*(double scalar) const
+    {
+      return {x * scalar, y * scalar, z * scalar, roll * scalar, pitch * scalar, yaw * scalar};
+    }
+  };
+  using Score = double;
+  struct Trial
+  {
+    Input input;
+    Score score;
+    int64_t id;
```

Comment:
`int64_t id;` is not used anywhere. please remove. :pray:

Reply:
Fixed when changing to generic TPE.